### PR TITLE
Issue #2251: Accept lowercased proxy settings

### PR DESCRIPTION
**Pull request by svenstaro**

### CODE REVIEW

**Comment by svenstaro** [comment]
AppVeyor failed due some unrelated error btw.

**Comment by gbraad** [comment]
AppVeyor mostly fails with unrelated errors

File: pkg/util/proxy.go [hunk 1]
```diff
@@ -39,7 +39,10 @@ type ProxyConfig struct {
 // the corresponding environment variable is checked.
 func NewProxyConfig(httpProxy string, httpsProxy string, noProxy string) (*ProxyConfig, error) {
 	if httpProxy == "" {
-		httpProxy = os.Getenv("HTTP_PROXY")
+		httpProxy = os.Getenv("http_proxy")
+		if httpProxy == "" {
+			httpProxy = os.Getenv("HTTP_PROXY")
+		}
```
Comment (praveenkumar) [inline pkg/util/proxy.go]: :+1:

File: pkg/util/proxy.go [hunk 2]
```diff
@@ -48,7 +51,10 @@ func NewProxyConfig(httpProxy string, httpsProxy string, noProxy string) (*Proxy
 	}
 
 	if httpsProxy == "" {
-		httpsProxy = os.Getenv("HTTPS_PROXY")
+		httpsProxy = os.Getenv("https_proxy")
+		if httpsProxy == "" {
+			httpsProxy = os.Getenv("HTTPS_PROXY")
+		}
```
Comment (praveenkumar) [inline pkg/util/proxy.go]: :+1:

File: pkg/util/proxy_test.go [hunk 3]
```diff
@@ -144,3 +146,19 @@ func Test_http_proxy_from_env(t *testing.T) {
 		os.Clearenv()
 	}
 }
+
+func Test_http_proxy_from_env_lowercase_precedence(t *testing.T) {
+	os.Clearenv()
+	defer os.Clearenv()
```
Comment (praveenkumar) [inline pkg/util/proxy_test.go]: :+1:

**Comment by svenstaro** [comment]
Looks like this is good to go. Can we merge this?

**Pull request merged**